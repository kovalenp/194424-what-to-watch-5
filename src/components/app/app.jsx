import React from "react";
import {Router, Route, Switch} from "react-router-dom";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import browserHistory from "../../common/browser-history";
import PrivateRoute from "../../common/private-route";
import Main from "../main/main";
import Film from "../film/film";
import SignIn from "../sign-in/sign-in";
import MyList from "../my-list/my-list";
import AddReview from "../add-review/add-review";
import Player from "../player/player";
import NotFound from "../not-found/not-found";
import {movieProps} from "../../validation/propTypes";
import {AppRoute} from "../../common/constants";
import withMovie from "../../hocs/with-movie/with-movie";

const AddReviewWithMovie = withMovie(AddReview);
const PlayerWithMovie = withMovie(Player);
const FilmWithMovie = withMovie(Film);

const App = (props) => {

  const {movies} = props;

  return (
    <Router history={browserHistory}>
      <Switch>
        <Route path={AppRoute.HOME} exact>
          <Main />
        </Route>
        <Route path={AppRoute.LOGIN} exact component={SignIn} />
        <PrivateRoute
          exact
          path={AppRoute.MY_LIST}
          render={() => <MyList movies={movies} />}
        />
        <PrivateRoute
          exact
          path={AppRoute.REVIEW}
          render={({match}) => (<AddReviewWithMovie
            id={match.params.id} />
          )}
        />
        <Route
          path={AppRoute.FILM}
          exact
          render={({match}) => (
            <FilmWithMovie
              id={match.params.id} />
          )}
        />
        <Route path={AppRoute.PLAYER} exact render={({match}) => (
          <PlayerWithMovie
            id={match.params.id} />
        )} />
        <Route component={NotFound} />
      </Switch>
    </Router>
  );
};

App.propTypes = {
  movies: PropTypes.arrayOf(movieProps),
};

const mapStateToProps = (state) => {
  return {
    movies: state.MOVIES.list,
  };
};

export {App};
export default connect(mapStateToProps, {})(App);
