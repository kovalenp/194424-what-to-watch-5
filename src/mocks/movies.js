export default [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    year: 2004,
    genre: `Drama`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 2,
    title: `Moonrise Kingdom`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/moonrise-kingdom.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 3,
    title: `Snatch`,
    year: 2004,
    genre: `Action`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/snatch.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 4,
    title: `Pulp Fiction`,
    year: 2004,
    genre: `Action`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/pulp-fiction.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 5,
    title: `No Country For Old Men`,
    year: 2004,
    genre: `Drama`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/no-country-for-old-men.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 6,
    title: `Bohemian Rhapsody`,
    year: 2004,
    genre: `Musical`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/bohemian-rhapsody.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 7,
    title: `Revenant`,
    year: 2004,
    genre: `Drama`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/revenant.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 8,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 9,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 10,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 11,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 12,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 13,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 14,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 15,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 16,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 17,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 18,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
  {
    id: 19,
    title: `Dardjeeling Limited`,
    year: 2004,
    genre: `Comedy`,
    director: `Wes Anderson`,
    cast: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    img: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      tile: `img/dardjeeling-limited.jpg`,
    },
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    plot: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.\n Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    rating: 8.9,
  },
];
