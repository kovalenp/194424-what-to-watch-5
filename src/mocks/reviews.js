export default [
  {
    "id": 1,
    "user": {"id": 1, "name": `pavel`},
    "rating": 5,
    "comment": `HTMLAcademy B21 criteria is my favorite`,
    "date": `2020-11-15T19:26:30.670Z`
  },
];
