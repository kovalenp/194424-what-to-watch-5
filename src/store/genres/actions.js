export const GenresActionsTypes = {
  SET_ACTIVE_GENRE: `SET_ACTIVE_GENRE`,
  GET_ALL_GENRES: `GET_ALL_GENRES`,
};

export const setGenre = (genre) => ({type: GenresActionsTypes.SET_ACTIVE_GENRE, payload: genre});
export const getGenresList = (movies) => ({type: GenresActionsTypes.GET_ALL_GENRES, payload: movies});
