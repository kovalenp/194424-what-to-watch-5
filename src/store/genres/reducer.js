import { GenresActionsTypes } from "./actions";
import movies from "../../mocks/movies";
import { ALL_GENRES } from "../../common/constants";

const initState = {
  activeGenre: ALL_GENRES,
  list: []
};

const getUniqueGenres = () => {
  const uniqueGenres = movies
    .map((movie) => movie.genre)
    .filter((value, index, self) => self.indexOf(value) === index);
  return [ALL_GENRES].concat(uniqueGenres);
};

const reducer = (state = initState, action) => {
  switch (action.type) {
    case GenresActionsTypes.SET_ACTIVE_GENRE:
      return Object.assign({}, state, { activeGenre: action.payload });
    case GenresActionsTypes.GET_ALL_GENRES:
      return Object.assign({}, state, {list: getUniqueGenres()});
    default:
      return state;
  }
};

export default reducer;
