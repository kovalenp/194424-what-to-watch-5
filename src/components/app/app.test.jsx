import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Provider} from "react-redux";
import movie from "../../mocks/movie";
import {activeGenre} from "../../mocks/genres";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  MOVIES: {current: movie, promo: movie, list: [movie]},
  USER: {authentication: `AUTH`},
  GENRES: {activeGenre}
});

jest.mock(`../main/main`, () => ({
  __esModule: true,
  default: () => () => <p>Main</p>,
}));

jest.mock(`../film/film`, () => ({
  __esModule: true,
  default: () => () => <p>Film</p>,
}));

jest.mock(`../../hocs/with-movie/with-movie`, () => ({
  __esModule: true,
  default: () => () => <p>withMovie</p>,
}));

jest.mock(`../movie-catalog/movie-catalog`, () => ({
  __esModule: true,
  default: () => () => <p>MovieCatalog</p>,
}));

describe(`App meets B21`, () => {

  afterAll(() => {
    jest.clearAllMocks();
  });

  it(`and rendered correctly`, () => {
    const tree = renderer.create((
      <Provider store={store}>
        <App />
      </Provider>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
