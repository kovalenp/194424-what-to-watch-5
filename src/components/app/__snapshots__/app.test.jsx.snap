// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App meets B21 and rendered correctly 1`] = `null`;
